subxt/src/error/dispatch_error.rs

```diff
@@ -142,6 +142,22 @@ impl PartialEq for ModuleError {
 
 impl Eq for ModuleError {}
 
+/// Custom `Debug` implementation, ignores the very large `metadata` field, using it instead (as
+/// intended) to resolve the actual pallet and error names. This is much more useful for debugging.
+impl Debug for ModuleError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let Ok(details) = self.details() else {
+            return f
+                .write_str("Unknown pallet error (pallet and error details cannot be retrieved)");
```

Comment:
Maybe better to have?:

```rust
write!(f, "ModuleError(<Unknown: pallet and error details cannot be retrieved>)")
```

So that it plays well if it ends up as part of a larger Debug call with other types etc involved

Reply:
I might also be tempted to havea similar format below like:

```
write!(f, "ModuleError(<{pallet_name}::{variant_name}>")
```

To be consistent with above and be clear that we aren't returning the internal struct layout, and instead something a bit custom? :)

Reply:
(But thank you; I love this! The default debug impl was stupid :D

Reply:
In both cases I would still expect to see the `bytes` though? How would that look?

Reply:
I've just added the `bytes` to the error in case the details cannot be resolved.

Reply:
Sorry for the slow reply; was off on a team retreat! Looks great to me!